provider/kubernetes/weight_allocator.go

```diff
@@ -0,0 +1,163 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/containous/traefik/provider/label"
+	"gopkg.in/yaml.v2"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+)
+
+type weightAllocator interface {
+	getWeight(host, path, serviceName string) int
+}
+
+var _ weightAllocator = &defaultWeightAllocator{}
+var _ weightAllocator = &fractionalWeightAllocator{}
+
+type defaultWeightAllocator struct{}
+
+func (d *defaultWeightAllocator) getWeight(host, path, serviceName string) int {
+	return label.DefaultWeight
+}
+
+type ingressService struct {
+	host    string
+	path    string
+	service string
+}
+
+type fractionalWeightAllocator struct {
```

Comment:
Couldn't `fractionalWeightAllocator` be typed as `map[ingressService]int` directly?